### DISCUSSION TITLE
Feature Request: ULID support

### DISCUSSION THREAD

**Discussion by liquidcarbon** [duckdb/duckdb#11853] [Ideas]
ULIDs are awesome as primary keys and in general.

It would be great to create utility functions around generating and converting ULIDs.

Examples:
- Postgres https://github.com/pksunkara/pgx_ulid
- Python https://github.com/itsrainingmani/py-ulid

I attempted using existing functions but the timestamp is encoded in base32, and there seems to be no way to decode it in pure DuckDB SQL.

**Comment by aag1024** [comment]
+1 Another advantage of ULID's is that they are a much better fit for DuckDB's default zonemap/min-max indexing than random UUID's.

**Comment by Maxxen** [comment]
Hello! I've built a [ULID extension here](https://github.com/Maxxen/duckdb_ulid) and also submitted it to the duckdb community repository so it should hopefully be installable from within duckdb soon.

**Comment by tonne1** [comment]
Since UUIDv7 is now official, support for ULID might be obsolete, pls see this discussion over at ULID https://github.com/ulid/spec/issues/91

**Reply by liquidcarbon** [reply]
Not sure how ULIDs are rendered obsolete by this RFC.  Are sequential integer IDs obsolete?  Different applications, different solutions...

UUID and ULID both start as 128 bits but in practice UUIDs inevitably end up as raw 36 char strings, annoying AF. ULIDs are shorter by 10 chars and still mean something.
